Lib/asyncio/runners.py

```diff
@@ -121,13 +121,7 @@ def run(self, coro, *, context=None):
             return self._loop.run_until_complete(task)
         except exceptions.CancelledError:
             uncancel = getattr(task, "uncancel", None)
-            if uncancel is None:
-                def uncancel():
-                    warnings._deprecated('missing asyncio.Task.uncancel',
-                        "Task.uncancel method will be mandatory for task "
-                        "implementations in 3.15", remove=(3, 15))
-                    return 0
-            if self._interrupt_count > 0 and uncancel() == 0:
+            if self._interrupt_count > 0 and uncancel is not None and uncancel() == 0:
```

Comment:
It seems we're doing unnecessary work if the interrupt count is zero -- we still ask the task for an uncancel method but never call it.

Reply:
done